pkg/prometheus/agent/operator.go

```diff
@@ -550,13 +551,21 @@ func (c *Operator) addHandlers() {
 }
 
 // Resolve implements the operator.Syncer interface.
-func (c *Operator) Resolve(ss *appsv1.StatefulSet) metav1.Object {
-	key, ok := c.accessor.MetaNamespaceKey(ss)
+func (c *Operator) Resolve(obj interface{}) metav1.Object {
+	key, ok := c.accessor.MetaNamespaceKey(obj)
 	if !ok {
 		return nil
 	}
 
-	match, promKey := statefulSetKeyToPrometheusAgentKey(key)
+	var match bool
+	var promKey string
+
+	if c.daemonSetFeatureGateEnabled {
```

Comment:
we also need to cast the given object to know if we should call the statefulset or daemonset function.